Dangerfile

```diff
@@ -5,90 +5,75 @@ if !defined? @platform
-# Node files
-files_to_check += ["yarn.lock", "docker-compose.yml", "Procfile", "npm-shrinkwrap.json", "node_modules", "tasks/options/env.coffee", "tslint.json"]
-# iOS files
-files_to_check += ["Cakefile", "fastlane/settings.yml.erb", "fastlane/Fastfile", "Podfile.lock"]
-# Check if files were modified
-(modified_files & files_to_check).each do |file|
-  message("`#{file}` modified")
+def checkForFileNode(file)
+  checkForNpmInstallGlobal(file)
+  checkForEnginesVersion(file)
+  validateSpecificExtensions(file)
 end
 
+def checkForFileIos(file)
+  checkTodo(file)
+  validateSwiftFiles(file)
+end
 
-# Warn if 'Gemfile' was modified and 'Gemfile.lock' was not
-if modified_files.include?("Gemfile")
-  if !modified_files.include?("Gemfile.lock")
-    warn("`Gemfile` was modified but `Gemfile.lock` was not")
-  else
-    message("`Gemfile` modified")
+def checkForFileAndroid(file)
+  ext = File.extname(file)
+  case ext
+  # Warn when a file .gradle is modified
+  when ".gradle"
+    message("`#{file}` was modified")
   end
+  # Warn when a FileManifest.xml is modified
+  message("`#{file}` was modified") if file =~ /Manifest\.xml/
 end
 
-modified_files.each do |file|
-  begin
-    ignoreWarn = !!(file =~ /.*\/(drawable|layout)\/.*/)
-    fileDiff = git.diff_for_file(file)
-    # Ensure we keep using secure https:// references instead of http://
-    httpMatches = fileDiff.patch.scan(/\+.*http:\/\/.*/)
-    httpMatches.each do |httpMatch|
-      warn("Detected unsecure `http://` use in `#{file}` section `#{httpMatch}`") if httpMatch && !ignoreWarn
-    end
+def checkForFileWeb(file)
+  checkForEnginesVersion(file)
+  checkForNpmInstallGlobal(file)
+  validateSwiftFiles(file)
+end
 
-    File.foreach(file) do |line|
-      line = line.gsub('\n','').strip
-      # Make sure resolves merges or rebases conflict issues
-      fail("Commited file without resolving merges/rebases conflict issues on `#{file}` - `#{line}`") if line =~ /^>>>>>>>/
-      # Look for Amazon Secret keys in modified files
-      warn("Possible amazon secret key hardcoded found in `#{file}` - `#{line}`") if line =~ /(?<![A-Za-z0-9\/+=])[A-Za-z0-9\/+=]{40}(?![A-Za-z0-9\/+=])/ && file != "yarn.lock" && file != "Podfile.lock"
-    end
-  rescue
-    message "Could not read file #{file}, does it really exist?"
-  end
+def checkForRegex(file, regex)
+  fileDiff = git.diff_for_file(file)
+  resultMatches = fileDiff.patch.scan(regex)
+  resultMatches
 end
 
 ########################
-#   FUNCTIONS SECTION  #
+#    NODE FUNCTIONS    #
 ########################
 
 def checkForEnginesVersion(file)
   # Keep engines version synced between declarations
   if file == "package.json"
-    fileDiff = git.diff_for_file(file)
+    # fileDiff = git.diff_for_file(file)
     engines = ["node","npm","yarn"]
     engines.each do |engine|
-      engineVersionMatches = fileDiff.patch.scan(/\+.*(\"#{engine}\")/)
+      engineVersionMatches = checkForRegex(file, /\+.*(\"#{engine}\")/)
```

Comment:
Which one is a false positive in this PR? Or you are talking about the `missing files` messages? I tried to run danger locally but this commit is already gone and can't be found :pensive:. I tried in another PR https://github.com/indigotech/br-pro120-jogo_das_forcas-server-api/pull/25

Result with latest release (1.1.3):
```
Messages:
- [ ] Could not read file src/api/admin/card.test.ts, does it really exist?
- [ ] Could not read file src/api/admin/context.test.ts, does it really exist?
- [ ] Could not read file src/api/admin/license.test.ts, does it really exist?
- [ ] Could not read file src/api/admin/registration.test.ts, does it really exist?
- [ ] Could not read file src/api/admin/user.test.ts, does it really exist?
- [ ] Could not read file src/api/card.test.ts, does it really exist?
- [ ] Could not read file src/api/context.test.ts, does it really exist?
- [ ] Could not read file src/domain/card-crud.use-case.ts, does it really exist?
- [ ] Could not read file src/domain/card-list.use-case.ts, does it really exist?
- [ ] Could not read file src/test/index.ts, does it really exist?
```
Result with this branch:
```
Warnings:
- [ ] Possibly returning 'any' in a function, prefer having a strongly typed return. `src/api/card.test.ts` at line `+  async register(app: any): Promise<any>`.
- [ ] Possibly returning 'any' in a function, prefer having a strongly typed return. `src/api/context.test.ts` at line `+  async register(app: any): Promise<any>`.
- [ ] Possibly returning 'any' in a function, prefer having a strongly typed return. `src/domain/card-list.use-case.ts` at line `+  async execute(pagination: PaginationModel, request: CardListRequest): Promise<any>`.
```